### PR TITLE
Update Item scroll component for load more item

### LINKED ISSUE #326: transactions: Cannot scroll up after scrolling down.
On master, when looking through transactions, after scrolling to the bottom of the list, cannot scroll back up:

https://github.com/crypto-power/cryptopower/assets/34998433/cdfd1781-26ab-4a7a-b590-a75239fce46c

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #326 
This PR Add new logic for items scroll component

### CODE REVIEW

**Review by dreacot — APPROVED** [review]

**Review by dreacot — APPROVED** [review]

File: ui/page/components/items_scroll.go [hunk 1]
```diff
@@ -78,102 +87,137 @@ func (s *Scroll[T]) FetchScrollData(isScrollUp bool, window app.WindowNavigator,
...
-		s.list.Position.OffsetLast = 1
-		s.offset -= s.pageSize
-	} else {
-		s.list.Position.Offset = 1
-		s.list.Position.OffsetLast = s.scrollView*-1 + 1
-		if s.data != nil {
-			s.offset += s.pageSize
+	s.mu.Lock()
+	tempSize := s.pageSize
+	//Scroll up and down without load more
```
Comment (dreacot) [inline ui/page/components/items_scroll.go:124]:
```suggestion
	// Scroll up and down without load more
```

File: ui/page/components/items_scroll.go [hunk 2]
```diff
@@ -24,6 +24,12 @@ const (
 	up   scrollDerection = 2
 )
 
+type dataList[T any] struct {
+	items    []T
+	idxStart int
```
Comment (dreacot) [inline ui/page/components/items_scroll.go:29]: what is idxStart

File: ui/page/components/items_scroll.go [hunk 3]
```diff
@@ -78,102 +87,137 @@ func (s *Scroll[T]) FetchScrollData(isScrollUp bool, window app.WindowNavigator,
...
-	s.isLoadingItems = true
-	itemsCountTemp := s.itemsCount
-	if s.itemsCount == -1 {
-		itemsCountTemp = 0
 	}
-	s.itemsCount = -1 // should trigger loading icon
-	offset := s.offset
-	tempSize := s.pageSize
 
+	// handle when need to load more items.
```
Comment (dreacot) [inline ui/page/components/items_scroll.go:164]:
```suggestion
	// handle when there is a need to load more items.
```

File: ui/page/components/items_scroll.go (hunk 2 above)
Comment (JustinBeBoy) [reply ui/page/components/items_scroll.go:29]: idxStart and edxEnd is a current index of items visible on list scroll, they are used to identify items to display in cache items

File: ui/page/components/items_scroll.go [hunk 4]
```diff
@@ -61,6 +69,7 @@ func NewScroll[T any](load *load.Load, pageSize int32, queryFunc ScrollFunc[T])
 		load:           load,
 		pageSize:       pageSize,
 		queryFunc:      queryFunc,
+		listSize:       pageSize * 2,
```
[the remark this answers is not in the document]
Comment (JustinBeBoy) [reply ui/page/components/items_scroll.go:72]: Idea is when display 2-pagesize items on list, we can append and remove items that keep the scroll indicator from moving to the top or bottom, it always stays at the bottom so the user can still move down or up again after moving down or up

**Review by dreacot — APPROVED** [review]

**Pull request merged**